editmsg.c

```diff
@@ -118,7 +122,22 @@ static int edit_one_message(struct Context *ctx, struct Header *cur)
     goto bail;
   }
 
-  mtime = mutt_decrease_mtime(tmp, &sb);
+  /* remove write permissions */
+  if (editOrView == VIEW)
+  {
+    /* Note: glibc has an ALLPERMS constant */
+    mode_t allperms = S_ISUID | S_ISGID | S_ISVTX | S_IRWXU | S_IRWXG | S_IRWXO;
+    mode_t perms = sb.st_mode & allperms; /* get only bits related to permissions */
+    rc = chmod(tmp, perms & ~(S_IWUSR | S_IWGRP | S_IWOTH));
+    if (rc == -1)
+    {
+      mutt_error(_("Could not remove write permissions of %s: %s"), tmp, strerror(errno));
+      goto bail;
+    }
+  }
```

Comment:
I'm not sure you really need this part. Most editors (well, vim at leat) will allow you to write the file anyway since the directory it's in belongs to the user and is writable. You are correctly checking `editOrView == VIEW && sb.st_mtime != mtime` below, so I'd say this just adds complexity.

Reply:
I'm unsure what you mean by "this part" (as the snippet only shows 4 lines):
1) bailing out.
2) the error message + bailing out.
3) marking the file as read-only.

In case 1) and 2). I can remove that since it is not essential to ensure the file is read-only (because---as you mentioned---we check for modification later / do not allow editing anyway).

In case 3): It is true that some editors allow you to write a read-only file. However, no editor AFAIK allows you to do it by accident. You have to explicitly request it (e.g. in vim with a bang!). This gives the user (hopefully) a strong indication that they are doing something which will not work as they expect.

Also most editors show some visual clue that the current file is read-only. This gives the user an additional clue that they are now in the 'view' mode and not in the 'edit' mode. This might be handy if you've forgotten whether your mailbox was read-only (remember `e` is now bound to `<edit-or-view-raw-message>`).

Though, I understand your point with the complexity.